test/factories/docker_tag_factory.rb

```diff
@@ -3,21 +3,20 @@
     sequence(:name) { |n| "2.#{n}" }
     repository :docker_repository
     association :docker_taggable, :factory => :docker_manifest
-  end
-  trait :schema1 do
-    after(:build) do |tag|
-      tag.docker_taggable.schema_version = 1
+
+    trait :schema1 do
+      association :docker_taggable, :factory => [:docker_manifest, :schema1]
+    end
+    trait :latest do
+      name "latest"
     end
-  end
-  trait :latest do
-    name "latest"
-  end
 
-  trait :with_uuid do
-    uuid { SecureRandom.hex }
-  end
+    trait :with_uuid do
+      uuid { SecureRandom.hex }
+    end
 
-  trait :with_manifest_list do
-    association :docker_taggable, :factory => :docker_manifest_list
+    trait :with_manifest_list do
+      association :docker_taggable, :factory => :docker_manifest_list
+    end
   end
```

Comment:
all these traits were OUTSIDE the ` factory :docker_tag` block, making them global.  I'm very certain this was not intentional.